src/main/java/com/google/devtools/build/lib/actions/ResourceSet.java

```diff
@@ -41,12 +44,24 @@
   /** The number of CPUs, or fractions thereof. */
   private final double cpuUsage;
 
+  /** Map of extra resources mapping name of the resource to a value. */
+  private final Map<String, Float> extraResourceUsage;
+
   /** The number of local tests. */
   private final int localTestCount;
 
   private ResourceSet(double memoryMb, double cpuUsage, int localTestCount) {
+    this(memoryMb, cpuUsage, null, localTestCount);
+  }
+
+  private ResourceSet(double memoryMb, double cpuUsage, Map<String, Float> extraResourceUsage, int localTestCount) {
     this.memoryMb = memoryMb;
     this.cpuUsage = cpuUsage;
+    if (extraResourceUsage == null) {
+        this.extraResourceUsage = new HashMap<>();
```

Comment:
Use Collections.emptyMap() instead.

Reply:
In the latest version I converted the extraResourceUsage from Map to an ImmutableMap, and using `ImmutableMap.of()` here.